sympy/controlsys/lti.py

```diff
@@ -0,0 +1,1363 @@
+                                 (self.__class__, name))
+
+        return handler
+
+    #
+    # _repr_latex_(self)
+    #   defines the representation of the class in ipython pretty printing
+    #
+    def _repr_latex_(self):
+        return '$' + latex(self._blockrepresent) + '$'
+
+    def __str__(self):
+        return 'StateSpaceModel(\n' + sstr(self.represent[0]) + ',\n' \
+                                    + sstr(self.represent[1]) + ',\n' \
+                                    + sstr(self.represent[2]) + ',\n' \
+                                    + sstr(self.represent[3]) + ')'
+
+    def __repr__(self):
+        return sstr(self)
+
+
+class TransferFunctionModel(object):
+    """ Transfer function model of a linear, time invariant control system
+
+    Represents the transfer Function model with a transfer function Matrix G in Laplace space.
+    The input-output relation for the system in Laplace space is then given by:
+        y(s) = G(s) * u(s);     s in C
+    where u(s) is the input of the system in Laplace space and y(s) the corresponding output
+
+    Parameters
+    ==========
+
+    arg : StateSpaceModel, Matrix
+        the state space model to construct the transfer function model from, or the transfer matrix G
+    s = None : Symbol
+        the variable G is dependent from. only has to be set if arg is a non-constant matrix or StateSpaceModel
+
+    See Also
+    ========
+
+    TranferFunctionModel: transfer function model of a lti system
+    Utils: mixed matrix and polynomial tools
+
+    References
+    ==========
+
+    Joao P. Hespanha, Linear Systems Theory. 2009.
+    """
+
+    def __init__(self, arg, s=None):
+
+        # check if a variable is given, if not create a new one as class-wide variable
+        if s:
+            self.s = s
+        else:
+            self.s = var('s')
+
+        # constructor from a given state space model
+        if isinstance(arg, StateSpaceModel):
+
+            try:
+                # define G as transfer function for the given state space model via the definition
+                self.G = arg.represent[2] * \
+                    (self.s * eye(arg.represent[0].shape[0]) - arg.represent[0]).inv() * \
+                    arg.represent[1] + arg.represent[3]
+
+                # try to simplify
+                self.G = simplify(self.G)
+
+            except ValueError as err:
+                raise ValueError(err.args, "Matrix (s*I -A) must be invertible")
+            except AttributeError:
+                raise TypeError("Only explicit Matrix Type supported for A,B,C,D (.inv() must work)")
+
+        # constructor from a given transfer function
+        elif isinstance(arg, (Matrix, ImmutableMatrix, MutableMatrix)):
+
+            # set the given transfer function as self.G
+            self.G = arg
+
+        else:
+            raise TypeError("argument of unsupported type")
+
+    #
+    # evaluate(self, u, s)
+    #
+    def evaluate(self, u, s):
```

Comment:
`evaluate()` seems like the wrong word for this operation to me. I'm not thinking of what the right word is right now, but I'll note it when I think of it.

Reply:
> `evaluate()` seems like the wrong word for this operation to me

May be we can rename this as `subs` and after substituting the value we can use `evalf` on the new object. What do you say?

Reply:
@moorepants `subs` seems fine to me. If you have a better name, please suggest cause I need to include this in my application!

Reply:
But the thing is that `StateSpaceModel` will subclass `Basic`. There's a method `subs` already.